kodi_addon_checker/check_files.py

```diff
@@ -139,3 +140,27 @@ def check_file_permission(report: Report, file_index: list):
         file = os.path.join(file["path"], file["name"])
         if os.path.isfile(file) and os.access(str(file), os.X_OK):
             report.add(Record(PROBLEM, "%s is marked as stand-alone executable" % relative_path(str(file))))
+
+
+def check_print_statement(report: Report, file_index: list):
+    """Check whether any addon files have a print statement in them
+       or not
+        :file_index: list having names and path of all the files present in addon
+    """
+    for file in file_index:
+        if os.path.splitext(file["name"])[1] == '.py':
+            file = os.path.join(file["path"], file["name"])
+
+            with open(file) as f:
+                source = f.read()
+
+            try:
+                for node in ast.walk(ast.parse(source)):
+                    if (isinstance(node, ast.Expr) and
+                            isinstance(node.value, ast.Call) and
+                            isinstance(node.value.func, ast.Name)):
+
+                        if node.value.func.id == "print":
+                            report.add(Record(WARNING, "%s has a print statement" % relative_path(str(file))))
+            except SyntaxError:
```

Comment:
SyntaxError gives a description and line number of the error, so that should reported.
```suggestion
            except SyntaxError as e:
```